File: Data Structures & Algorithms/find-anagram-mappings/submission-1.java
class Solution {

    public int[] anagramMappings(int[] nums1, int[] nums2) {
        Map<Integer, Integer> mapping = new HashMap<>();

        for (int i = 0; i < nums2.length; i++) {
            mapping.put(nums2[i], i);
        }

        int[] res = new int[nums1.length];

        for (int i = 0; i < nums1.length; i++) {
            res[i] = mapping.get(nums1[i]);
        }

        return res;
    }
}

File: Data Structures & Algorithms/count-subsequences/submission-1.py
class Solution:
    def numDistinct(self, s: str, t: str) -> int:
        if len(t) > len(s):
            return 0

        memo = {}

        def dfs(i, j):
            if j == len(t):
                return 1
            if i == len(s):
                return 0

            if (i, j) in memo:
                return memo[(i, j)]

            res = dfs(i + 1, j)
            if s[i] == t[j]:
                res += dfs(i + 1, j + 1)
            memo[(i, j)] = res
            return res
            
        return dfs(0, 0)